crates/uv/src/commands/tool/run.rs

```diff
@@ -214,7 +216,28 @@ pub(crate) async fn run(
     // signal handlers after the command completes.
     let _handler = tokio::spawn(async { while tokio::signal::ctrl_c().await.is_ok() {} });
 
-    let status = handle.wait().await.context("Child process disappeared")?;
+    let mut term_signal = signal(SignalKind::terminate())?;
+    let mut int_signal = signal(SignalKind::interrupt())?;
+
+    let status = select! {
+        status = handle.wait() => status,
+
+        // `SIGTERM`
+        _ = term_signal.recv() => {
+            handle.kill().await?;
+            handle.wait().await.context("Child process disappeared")?;
+            return Ok(ExitStatus::Failure);
+        }
+
+        // `SIGINT`
+        _ = int_signal.recv() => {
+            handle.kill().await?;
```

Comment:
Wouldn't it be important to preserve the signal kind, so still a SIGINT to the child?